guides/puppet-web-apps/authorization/README.md

```diff
@@ -0,0 +1,227 @@
+* PersonClaim - if present it means that a current user represents specific person. It is preferred mean of establishing user's identity, since it's usually more convenient to manage People than Users
+
+There are other built-in rules in `Starcounter.Authorization.Core.Rules` namespace. Feel free to use them or create your own.
+
+### Check the rules
+
+`AuthorizationEnforcement` is a class responsible for checking the if the current user has a specific permission with regard to a rule set. Let's create it using the rules we defined in previous steps:
+
+```c#
+var enforcement = AuthorizationEnforcement(rules, new SystemUserAuthentication());
+```
+
+The second argument passed is a authentication backend (a something that obtains the claims about a current user). It's usually enough to use the built in `SystemUserAuthentication`. It will provide `SystemUserClaim` and `PersonClaim` for singed in users.
+
+We can now use our new `IAuthorizationEnforcement` to check if we have a permission:
+
+```c#
+bool canWe = enforcement.CheckPermission(new ListInvoices()); // note that the outcome depends on weather we are currently signed in as "admin"
+```
+
+## Automatic checking of permissions - SecurityMiddleware
+
+Manual checking of permissions is powerful, but usually we need something that will check the permissions for us. That's what SecurityMiddleware is doing - it's preventing users from accessing pages they don't have access to and from making interactions they are not allowed to do.
+
+### Prerequisite: Router
+
+`SecurityMiddleware` is a feature that depends on Router. If you don't know what it is - go ahead and check it out // `TODO` LINK
+
+To enable `SecurityMiddleware` add it to a Router:
+
+```c#
+router.AddMiddleware(new SecurityMiddleware(
+enforcement,
+info => Response.FromStatusCode(403), // a function returning a Response to give instead of a restricted Page
+PageSecurity.CreateThrowingDeniedHandler<Exception>())); // defines the behavior in case of forbidden interaction
+```
+`TODO` describe checkDeniedHandler
+
+### Restricting access to a page
+
+Primary mean of managing access to pages will be the `RequiredPermissionAttribute`. Let's use it to mark secure our example page:
+
+```c#
+[Url("/invoices/invoices")]
+[RequirePermission(typeof(ListInvoices))]
+public partial class InvoicesPage : Json
+{
+// ...
+}
+```
+
+After adding this attribute, every time a user will open the URL "/invoices/invoices", the Authorization library will check weather he has is granted a ListInvoices Permission. If he is not, then he'll be presented with 403 Forbidden page (or any other response configured in `SecurityMiddleware` constructor). What's more, every time a user will try to change a property on this page or run a handler on this page, the same permission will be checked. In case of failure, an Exception will be thrown (this also can be customised using the `SecurityMiddleware` constructor).
```

Comment:
And also please remove the excessive usage of "will". Present simple is enough here.